lms/djangoapps/program_enrollments/api/v1/tests/test_views.py

```diff
@@ -171,3 +173,191 @@ def test_200_many_pages(self):
             assert next_response.data['previous'] is not None
             assert self.get_url(self.program_uuid) in next_response.data['previous']
             assert '?cursor=' in next_response.data['previous']
+
+
+class ProgramEnrollmentViewPostTests(APITestCase):
+    """
+    Tests for the ProgramEnrollment view POST method.
+    """
+
+    def test_successful_program_enrollments(self):
+        global_staff = GlobalStaffFactory.create(username='global-staff', password='password')
+        self.client.login(username=global_staff.username, password='password')
+        program_key = uuid4()
+        statuses = ['pending', 'enrolled', 'pending']
+        external_user_keys = ['abc1', 'efg2', 'hij3']
+
+        curriculum_uuid = uuid4()
+        curriculum_uuids = [curriculum_uuid, curriculum_uuid, uuid4()]
+        post_data = [
+            {
+                'external_user_key': e,
+                'status': s,
+                'curriculum_uuid': str(c)
+            }
+            for e, s, c in zip(external_user_keys, statuses, curriculum_uuids)
+        ]
+
+        url = reverse('programs_api:v1:program_enrollments', args=[program_key])
+        with mock.patch('lms.djangoapps.program_enrollments.api.v1.views.get_programs', autospec=True):
+            response = self.client.post(url, json.dumps(post_data), content_type='application/json')
+
+        self.assertEqual(response.status_code, 201)
+
+        for i in range(3):
+            enrollment = ProgramEnrollment.objects.filter(external_user_key=external_user_keys[i])[0]
+            actual_external_user_key = enrollment.external_user_key
+            self.assertEqual(actual_external_user_key, external_user_keys[i])
+
+            actual_program_uuid = enrollment.program_uuid
+            actual_status = enrollment.status
+            actual_curriculum_uuid = enrollment.curriculum_uuid
+
+            self.assertEqual(actual_program_uuid, program_key)
+            self.assertEqual(actual_status, statuses[i])
+            self.assertEqual(actual_curriculum_uuid, curriculum_uuids[i])
+
+    def test_enrollment_payload_limit(self):
+        global_staff = GlobalStaffFactory.create(username='global-staff', password='password')
+        self.client.login(username=global_staff.username, password='password')
+        def student_enrollment(enrollment_status):
```

Comment:
Everywhere this function is being used, the function signature and logic is largely the same. Refactor this function out into a class function like this:
```
def generate_student_enrollments_input(enrollment_status, external_user_key_override=None):
    return {
                'status': enrollment_status,
                'external_user_key': external_user_key_override or str(uuid4().hex[0:10]),
                'curriculum_uuid': str(uuid4())
            }
```

Reply:
Was planning on this refactor. Thanks!